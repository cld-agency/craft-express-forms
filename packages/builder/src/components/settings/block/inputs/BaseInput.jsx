import Tooltip from '@material-ui/core/Tooltip/Tooltip';
import React from 'react';
import PropTypes from 'prop-types';
import translate from '../../../../functions/translator';

class BaseInput extends React.Component {
  static propTypes = {
    label: PropTypes.string,
    description: PropTypes.string,
    required: PropTypes.bool,
    name: PropTypes.string,
    value: PropTypes.any,
    saveHandler: PropTypes.func,
    valueToCamelCase: PropTypes.bool,
    disabled: PropTypes.bool,
  };

  static defaultProps = {
    description: null,
    required: false,
  };

  getBlockClasses() {
    return [];
  }

  renderInput() {}

  render() {
    const { label, required, description } = this.props;

    const blockClasses = ['block-row', ...this.getBlockClasses()];

    return (
      <div className={blockClasses.join(' ')}>
        {label && (
          <div className="label">
            <Tooltip title={translate(description) ?? ''} placement="bottom">
              <label className={required ? 'required' : ''}>{translate(label)}</label>
            </Tooltip>
          </div>
        )}
        <div className="input ltr">{this.renderInput()}</div>
      </div>
    );
  }
}

export default BaseInput;
