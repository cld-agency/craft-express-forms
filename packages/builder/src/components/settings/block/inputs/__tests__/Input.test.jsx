import { shallow } from 'enzyme';
import React from 'react';
import BaseInput from '../BaseInput';

describe('<BaseInput />', () => {
  it('renders & displays correctly', () => {
    const wrapper = shallow(<TestInput label="Test Input" name="test_input" value="test value" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('shows asterisk if required', () => {
    const wrapper = shallow(<TestInput label="Test Input" required={true} name="test_input" value="test value" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('label.required')).toHaveLength(1);
  });
});

class TestInput extends BaseInput {
  static propTypes = {
    ...BaseInput.propTypes,
  };

  renderInput() {
    return <input type="text" value="test value" />;
  }
}
