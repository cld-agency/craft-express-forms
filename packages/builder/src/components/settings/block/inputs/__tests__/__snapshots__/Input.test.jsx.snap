// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseInput /> renders & displays correctly 1`] = `
<div
  className="block-row"
>
  <div
    className="label"
  >
    <WithStyles(ForwardRef(Tooltip))
      placement="bottom"
      title=""
    >
      <label
        className=""
      >
        Test Input
      </label>
    </WithStyles(ForwardRef(Tooltip))>
  </div>
  <div
    className="input ltr"
  >
    <input
      type="text"
      value="test value"
    />
  </div>
</div>
`;

exports[`<BaseInput /> shows asterisk if required 1`] = `
<div
  className="block-row"
>
  <div
    className="label"
  >
    <WithStyles(ForwardRef(Tooltip))
      placement="bottom"
      title=""
    >
      <label
        className="required"
      >
        Test Input
      </label>
    </WithStyles(ForwardRef(Tooltip))>
  </div>
  <div
    className="input ltr"
  >
    <input
      type="text"
      value="test value"
    />
  </div>
</div>
`;
